duckduckgo/Android
# Fix empty entry whitelist optional addon: Introduce Junit5 unit tests

Task/Issue URL: 
Tech Design URL: 
CC: 

**Description**:
same like PR https://github.com/duckduckgo/Android/pull/986 just with additional unit tests added. 
Adds [Junit5](https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions-third-party) test dependency which allows the use of ParameterisedTest which is really handy to test lots of scenarios in one test method.
Also removed some test dependencies that seemed unused (only used in androidTest)

**Note**: I have noticed that there wasn't a `test` source set added yet to your project, there was only an `androidTest` yet. So you might need to add another step to your CI to run `./gradlew testDebugUnitTest` to not miss this tests.

**Steps to test this PR**:
1. run ./gradlew testDebugUnitTest in terminal or use Android studio to run jvm unit tests by clicking `run test` icon next to `ListExtensionKtTest`
1. See green test results:
![Screenshot from 2020-11-01 11-44-07](https://user-images.githubusercontent.com/8261416/97801855-a01d8400-1c37-11eb-983f-de3544f9ddce.png)



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
